dotcom-rendering/src/web/layouts/FrontLayout.tsx

```diff
@@ -38,6 +38,14 @@ interface Props {
 	NAV: NavType;
 }
 
+const adStyles = css`
+	.ad-slot {
+		&.ad-slot--collapse {
+			display: none;
+		}
+	}
+`;
```

Comment:
Is there a reason this logic doesn't live within the adslot component itself?

Reply:
For articles we needed collapsing at a higher level to handle:
- ads inserted client side by spacefinder
- hardcoded ads in interactives

For fronts all slots are rendered server side so they could move to the ad slot. It would just mean duplicating collapsing css at a slot and container level for articles.

I think adding the collapsing css to the slots is a good idea if only to make it visible to developers working on `AdSlot`

Reply:
You might want to follow a pattern we use for Islands, [where we have global styles exported from where people are most likely to encouter them: `Island.tsx`](https://github.com/guardian/dotcom-rendering/blob/b75b73858f30d15d3be15ee57022a8264663eb38/dotcom-rendering/src/web/components/Island.tsx#L149-L156).

Reply:
Thanks all for your suggestions! I pushed some changes which I believe are aligned with the spirit of your comments.